packages/components/src/vue/vuex-modules/data/lexis.js

```diff
@@ -469,40 +465,62 @@ export default class Lexis extends Module {
       }
 
       if (wqData && wqData.homonym) {
-        // A lexical query will be used to retrieve data that is not served by GraphQL currently.
-        const lexQuery = LexicalQuery.create(textSelector, {
+        let homonym = wqData.homonym // eslint-disable-line prefer-const
+
+        // Get lemma translations. Lemma translations are enabled for Latin only
+        if (this._lemmaTranslationEnabled && textSelector.languageID === Constants.LANG_LATIN) {
+          homonym = await Lexis.getLemmaTranslations({
+            homonym, browserLang: this._lemmaTranslationLocale, clientId: this._appApi.clientId, source
+          })
+        }
+
+        // Get word usage examples
+        if (wordUsageExamples) {
+          homonym = await Lexis.getWordUsageExamples({
+            homonym, paginationAuthMax: wordUsageExamples.paginationAuthMax, clientId: this._appApi.clientId
+          })
+        }
+
+        const languageCode = LMF.getLanguageCodeFromId(textSelector.languageID)
+        homonym = await Lexis.getFullDefinitions({
+          homonym,
+          languageCode,
           clientId: this._appApi.clientId,
+          location: textSelector.location,
           siteOptions,
-          verboseMode: this._settingsApi.isInVerboseMode(),
-          lemmaTranslations,
-          wordUsageExamples,
-          resourceOptions: this._settingsApi.getResourceOptions(),
-          langOpts: { [Constants.LANG_PERSIAN]: { lookupMorphLast: true } }, // TODO this should be externalized
-          checkContextForward,
-          cedictServiceUrl: this.hasCedict() ? this._lexisConfig.cedict.target_url : null,
-          source,
-          hasLexemes: true,
-          homonym: wqData.homonym
+          resourceOptions: this._settingsApi.getResourceOptions()
         })
-        result = lexQuery.getData()
+        if (source !== LexicalQuery.sources.WORDLIST) {
+          LexicalQuery.evt.LEXICAL_QUERY_COMPLETE.pub({
+            resultStatus: LexicalQuery.resultStatus.SUCCEEDED,
+            homonym: homonym
+          })
+        }
       } else {
         Logger.getInstance().error('Homonym is not available')
-        if (this._source !== LexicalQuery.sources.WORDLIST) {
+        if (source !== LexicalQuery.sources.WORDLIST) {
           LexicalQuery.evt.LEXICAL_QUERY_COMPLETE.pub({
             resultStatus: LexicalQuery.resultStatus.FAILED
           })
         }
       }
     } else {
       /*
-      Use the old workflow for Chinese for now.
+      Use the old workflow for languages other than Latin and Greek
        */
       let annotatedHomonyms
       if (treebankWordIDs.length > 0) {
         annotatedHomonyms = await this.getTreebankData({
           store, textSelector, treebankDataItem
         })
       }
+
+      let lemmaTranslations
+      if (this._lemmaTranslationEnabled && textSelector.languageID === Constants.LANG_LATIN && this._lemmaTranslationLocale) {
+        // Use our own rules if lemmaTranslations is not provided
```

Comment:
this comment doesn't make sense to me.

Reply:
Oh, you're right! The code being commented is executed only if the text selector's language is Latin. This code block is, however, in the `else` clause that will be executed only if the language is not Latin: https://github.com/alpheios-project/alpheios-core/blob/i566-resource-retrieval/packages/components/src/vue/vuex-modules/data/lexis.js#L434. As a result, the content of the `if` statement will never be run. Because of this, I've decided to remove it.